routers/api/v1/teams.go

```diff
@@ -35,6 +35,53 @@ func (r *apiV1Router) GetTeams(ctx *gin.Context) {
 	})
 }
 
+// GET: /api/v1/teams/:id
+// Response: status int
+//           error string
+//           team entities.Team
+// Headers:  Authorization -> token
+func (r *apiV1Router) GetMyTeam(ctx *gin.Context) {
```

Comment:
Is this endpoint supposed to return the user's team or just a team with an ID? The name suggests that it would return the user's team but in reality it just gives you the team with the given ID.

It's also accessible to all registered users, while I wouldn't call this a security vulnerability because you'd need to guess the ID of a random team (which is quite unlikely given that it's a UUID), I still don't think that this kind of functionality (getting information about a team with given ID) should be available to any user.

Reply:
@seanjparker

Reply:
Ah right, so I added this for hs_hub. Basically, I didn't see an endpoint that would return the team info (name, table number, id etc) for the users team. The only endpoint I could of used was getAllTeams but it was only available to admins. I added this as a temporary solution for the hackathon that I would call it with the users team ID.

I agree that it shouldn't be accessible to be called with any team ID. Would it be possible to alter it to only get the team of the user based on the auth token?

Also, I guess we could keep a similar endpoint that is only accessible to  volunteers/admins?